docs/articles/viewing-epoch-based-metrics.rst

```diff
@@ -0,0 +1,232 @@
+************************
+ Step 2: Report Metrics
+************************
+
+In this section, we'll define our epoch metric. We'll also report training and validation metrics to
+the Determined master. To do this, we'll import Determined, and create a
+:class:`~determined.core.Context` object to allow interaction with the master. Then, we'll pass the
+``core_context`` as an argument into ``main()``, ``train()``, and ``test()`` and modify the function
+headers accordingly.
+
+To follow along, use the ``model_def_metrics.py`` script and its accompanying ``metrics.yaml``
+experiment configuration file.
+
+We'll start by importing Determined:
+
+.. code:: bash
+
+   python3
+
+.. code:: python
+
+   import determined as det
+
+Step 2.1: Verify the Main Loop Contains ``core.Context``
+========================================================
+
+We'll need a ``core.Context`` object for interacting with the master. Verify the __main__loop in
+your script, ``model_def_metrics.py``, contains ``core_context``:
+
+.. code:: python
+
+   if __name__ == "__main__":
+
+     # NEW: Establish new determined.core.Context and pass to main
+     # function.
+     with det.core.init() as core_context:
+         main(core_context=core_context)
+
+Step 2.2: Modify the Train and Validation Methods to Report Epoch-Based Metrics
+===============================================================================
+
+Our script already contains ``core_context.train``. This is used to report training and validation
+metrics.
+
+But, we also want to report epoch-based metrics. To do this, we'll modify the train() method to
+include ``epoch_idx`` as a metric. This allows Determined to keep track of the specific epoch for
+which training loss is being reported:
+
+.. code:: python
+
+   # NEW: Report epoch-based training metrics to Determined
+   # master via core_context.
+   # Index by (batch_idx + 1) * (epoch-1) * len(train_loader)
+   # to continuously plot loss on one graph for consecutive
+   # epochs.
+   core_context.train.report_training_metrics(
+       steps_completed=batches_completed + epoch_idx * len(train_loader),
+       metrics={"train_loss": loss.item(), "epoch": epoch_idx},
+
+   )
+
+Similarly, we'll include ``epoch`` as a metric in the reported validation metrics. This allows
+Determined to track the specific epoch for which the validation loss is being reported:
+
+.. code:: python
+
+   # NEW: Report epoch_based validation metrics to Determined master
+   # via core_context.
+   core_context.train.report_validation_metrics(
+       steps_completed=steps_completed,
+       metrics={"test_loss": test_loss, "epoch": epoch},
+
+   )
+
+Now that we've reported an epoch value, **Epoch** will be an available option for the X-Axis when we
+view our metric data graph in the WebUI.
+
+Step 2.3: Verify the Test Method
+================================
+
+We'll need a ``test()`` function to evaluate the trained model on the test/validation data for the
+current epoch.
```

Comment:
This step doesn't make much sense, because the presence of a step suggest that there's going to be some code changes, but it looks to me like the code calling `test()` is already correct?

Reply:
(also generally, I have similar confusion why we are asking a user to "verify" code they downloaded form us)

Reply:
mitigated as in previous comment